### ISSUE TITLE
Docker-desktop v4.17.1 Dev Environments recreating container on start

### ISSUE THREAD

**Issue by xenoliss** [docker/for-win#13352]
- [x ] I have tried with the latest version of Docker Desktop
  - [ x] I have tried disabling enabled experimental features
  - [x ] I have uploaded Diagnostics
  - Diagnostics ID:

### Actual behavior
(re)s tarting a docker devenv causes the containers to be recreated.

### Expected behavior
(re)s tarting a docker devenv should use the already created containers.

### Information
Please, help us understand the problem.  For instance:
  - Is it reproducible? YES
  - Is the problem new? YES
  - Did the problem appear with an update? Migrating from v15 to v17 caused the issue
  - Windows Version: 11
  - Docker Desktop Version: v4.17.1
  - WSL2 or Hyper-V backend?   WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`
```
[2023-04-01T18:36:18.143581800Z][com.docker.diagnose.exe][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0002: does the bootloader have virtualization enabled?
[SKIP] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[FAIL] DD0029: is the WSL 2 Linux filesystem corrupt? [ 3421.608906] EXT4-fs error (device sdd): ext4_put_super:1188: comm wsl-bootstrap: Couldn't clean up the journal
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0005: is the user in the docker-users group?
[PASS] DD0038: is the connection to Docker working?
[PASS] DD0014: are the backend processes running?
[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[WARN] DD0029: is the WSL 2 Linux filesystem corrupt? [ 3421.608906] EXT4-fs error (device sdd): ext4_put_super:1188: comm wsl-bootstrap: Couldn't clean up the journal
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?

Please note the following 1 warning:

1 : The check: is the WSL 2 Linux filesystem corrupt?
    Produced the following warning: [ 3421.608906] EXT4-fs error (device sdd): ext4_put_super:1188: comm wsl-bootstrap: Couldn't clean up the journal

If the WSL 2 Linux filesystem is corrupt then Docker Desktop cannot start.
There is a known issue fixed in Windows Insider builds which can cause filesystem corruption, see:
https://github.com/microsoft/WSL/issues/5895 .

Try running "wsl --shutdown" to stop your WSL Virtual Machine. When it restarts it will
run a filesystem check and hopefully fix the problem.



Please investigate the following 1 issue:

1 : The test: is the WSL 2 Linux filesystem corrupt?
    Failed with: [ 3421.608906] EXT4-fs error (device sdd): ext4_put_super:1188: comm wsl-bootstrap: Couldn't clean up the journal

If the WSL 2 Linux filesystem is corrupt then Docker Desktop cannot start.
There is a known issue fixed in Windows Insider builds which can cause filesystem corruption, see:
https://github.com/microsoft/WSL/issues/5895 .

Try running "wsl --shutdown" to stop your WSL Virtual Machine. When it restarts it will
run a filesystem check and hopefully fix the problem.
```

### Steps to reproduce the behavior

  1. Install docker-destkop v4.17.1
  2. Use the simple docker devenv example (with some go code) https://github.com/dockersamples/single-dev-env
  3. Start the devenv and open vscode within the container
  4. Install something like rust: `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
  5. Stop the dev-env **by clicking the stop icon of the devenv**
  6. Restart the dev-env **by clicking the start icon of the dev env**
  7. Re-open vscode within the container.
  8. Notice rust has been removed (aka, the container has been recreated)
  
  Basically the bug seems to be that clicking "start" icon of the devenv recreate the containers (also visible on the log files).
  When directly clicking on the container "start" icon, the container does not get recreated.
